contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -2381,10 +2381,16 @@ static void process_select_statement(
 	PLtsql_expr_query_mutator *mutator)
 {
 	/* remove unsupported_tokens */
-	if (selectCtx->for_clause()) // for xml
+	if (selectCtx->for_clause()) 
 	{
-		Assert(selectCtx->for_clause()->XML()); // we only support FOR XML
-		Assert(selectCtx->for_clause()->RAW() || selectCtx->for_clause()->PATH());
+		if (selectCtx->for_clause()->XML()) // FOR XML
+		{
+			Assert(selectCtx->for_clause()->RAW() || selectCtx->for_clause()->PATH());
+		}
+		else if (selectCtx->for_clause()->JSON()) // for JSON
+		{
+			Assert(selectCtx->for_clause()->PATH());
+		}		
```

Comment:
1. What about `else` case?
2. `Assert(selectCtx->for_clause()->PATH() || selectCtx->for_clause()->AUTO());`?

Reply:
I am not sure about the else case. Are you suggesting to add `Assert(selectCtx->for_clause()->PATH() || selectCtx->for_clause()->AUTO());` in the else case?

Reply:
Resolved in the latest commit